final_task/rss_reader/console_interface.py

```diff
@@ -0,0 +1,47 @@
+import argparse
+import logging
+
+class ConsoleInterface():
+    """ Implement console interface"""
+
+    source = None
+    limit = -1
+    version = "Version 1.4.1"
+    json = False
+    verbose = False
+    date = None
+    to_html = False
+    path = None
+    def __init__(self):
+        """ Gets parametres from console"""
+
+        parser = argparse.ArgumentParser(description="Pure Python command-line RSS reader.")
+        parser.add_argument("source", help="RSS url")
+        parser.add_argument("--limit", type=int, help="Limit news topics if this parameter provided")
+        parser.add_argument("--version", action="store_true", help="Print version info")
+        parser.add_argument("--json", action="store_true", help="Print result as JSON in stdout")
+        parser.add_argument("--verbose", action="store_true", help="Outputs verbose status messages")
+        parser.add_argument("--date", type=str, help="Print the new from the specified day, YYYYMMDD format")
+        parser.add_argument("--to-html", type=str,
+                            help="Convert news in html format, need path, where the file will be saved")
+        args = parser.parse_args()
+
+
+        self.source = args.source
+        if args.limit:
+            self.limit = args.limit
+        if args.version:
+            print(self.version)
+        if args.json:
+            self.json = True
+        if args.verbose:
+            self.verbose = True
+            logging.basicConfig(level=logging.DEBUG, format='%(process)d-%(levelname)s-%(message)s')
+        else:
+            logging.basicConfig(filename='app.log', level=logging.DEBUG, filemode='w', format='%(name)s - %(levelname)s - %(message)s')
+        if args.date:
+            self.date = args.date
+        logging.debug("Check in interface")
+        if args.to_html:
+            self.to_html = True
+            self.path = args.to_html
```

Comment:
You do not really need a whole class for this. It is better to use two functions instead: one which parses args and the other which validates them.

Reply:
Fixed